src/kernels/jupyter/launcher/notebookProvider.ts

```diff
@@ -53,7 +53,9 @@ export class NotebookProvider implements INotebookProvider {
             return this.jupyterNotebookProvider.connect(options).finally(() => handler.dispose());
         } else {
             handler.dispose();
-            await this.extensionChecker.showPythonExtensionInstallRequiredPrompt();
+            if (!this.startupUi.disableUI) {
```

Comment:
I have to confess I dug though the history a bit and I'm not sure why this wasn't being seen before? It does require somewhat specific circumstances (disableZMQ, no Python installed) but multiple vscode team members hit it pretty quick.

Reply:
The issue looks to my eyes like it has been around for quite a while.